### PR TITLE
Ros2 foxy

### PR DESCRIPTION
**Pull request by cheng-kevin**
*Issue #, if available:*

*Description of changes:*
Update to support ROS2 Foxy.  create_publisher was only one function that needed an update, as there has been a new required parameter added to it.  

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: robot_ws/src/hello_world_robot/src/hello_world_robot/rotate.py [hunk 1]
```diff
@@ -28,7 +28,9 @@ class Rotator(Node):
 
     def __init__(self):
         super().__init__('rotate')
-        self._cmd_pub = self.create_publisher(Twist, '/cmd_vel')
+        # For Foxy, qos_profile is a required field for create_publisher api
+        # For Dashing, qos_profile is optional and defaults to 10.
+        self._cmd_pub = self.create_publisher(Twist, '/cmd_vel', 10)
```
Comment (ojasjoshi) [inline robot_ws/src/hello_world_robot/src/hello_world_robot/rotate.py:33]: https://github.com/ros2/rclpy/blob/dashing/rclpy/rclpy/node.py#L1027 seems to suggest dashing takes optional qos_profile parameter. To be safe, have you tested this does not break dashing?

Comment (ojasjoshi) [reply robot_ws/src/hello_world_robot/src/hello_world_robot/rotate.py:33]: *I would suggest adding tests related description in the PR if possible

File: simulation_ws/.rosinstall [hunk 2]
```diff
@@ -1,2 +1 @@
 - git: {local-name: src/deps/turtlebot3-description-reduced-mesh, uri: "https://github.com/aws-robotics/turtlebot3-description-reduced-mesh.git", version: "ros2"}
-- git: {local-name: src/deps/gazebo-ros-pkgs, uri: "https://github.com/ros-simulation/gazebo_ros_pkgs.git", version: "c071749932a541a5b8aced23c3414724a8cd1949"}
```
Comment (ojasjoshi) [inline simulation_ws/.rosinstall:2]: Is this an unused dependency?

Comment (ojasjoshi) [inline simulation_ws/.rosinstall:2]: this is removed because it's pulled via package.xml?

**Comment by cheng-kevin** [comment]
ros2_foxy will be kept as a separate branch. Closing this PR

**Pull request closed without merging**